### PR TITLE
fix(ingest): Generate browse paths v2 for more sources; properly pass platform_instance

### PR DESCRIPTION
**Pull request by asikowitz**
Makes the following changes:
- Passes `platform_instance` properly as per https://github.com/datahub-project/datahub/pull/8478, thanks @anshbansal!
- Remove overrides of `get_workunits` in many sources so that they run workunit processors, which will make them both (i) create status aspects and (ii) create browse paths v2
- Updates error handling around MCPDiff -- does not log warning or print trace if file contains MCEs

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/integration/remote/golden/remote_enricher_golden.json [hunk 1]
```diff
@@ -158,5 +158,35 @@
...
+    "systemMetadata": {
+        "lastObserved": 1629795600000,
+        "runId": "remote-1"
+    }
+},
+{
+    "entityType": "tag",
+    "entityUrn": "urn:li:tag:Legacy",
+    "changeType": "UPSERT",
+    "aspectName": "tagKey",
```
Comment (asikowitz) [inline metadata-ingestion/tests/integration/remote/golden/remote_enricher_golden.json:181]: Getting a tag key aspect, just want to make sure this makes sense for this source

**Review by asikowitz — COMMENTED** [review]
I scanned thru all the goldens and changes looked reasonable to me. One question around the "remote" source -- which is the csv_enricher / datahub-lineage-file / datahub-business-glossary. Not sure whether we should be auto generating stuff for them or not

File: metadata-ingestion/src/datahub/ingestion/source/metadata/lineage.py [hunk 2]
```diff
@@ -139,8 +143,11 @@ def load_lineage_config(file_name: str) -> LineageConfig:
         lineage_config = LineageConfig.parse_obj(config)
         return lineage_config
 
-    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
-        return auto_workunit_reporter(self.report, self.get_workunits_internal())
+    def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
```
[the remark this answers is not in the document]
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/metadata/lineage.py:146]: Yep this makes sense

File: metadata-ingestion/src/datahub/ingestion/source/mongodb.py [hunk 3]
```diff
@@ -2,10 +2,6 @@
 from dataclasses import dataclass, field
 from typing import Dict, Iterable, List, Optional, Tuple, Type, Union, ValuesView
 
-import bson
-import bson.dbref
-import bson.int64
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/mongodb.py:7]: Were these just unused?

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/mongodb.py (hunk 3 above)
Comment (asikowitz) [reply metadata-ingestion/src/datahub/ingestion/source/mongodb.py:7]: Ah I just ran the intellij "optimize imports" command on a bunch of files had imports to delete. It's very possible these are needed, we'll see what CI says I guess

**Comment by anshbansal** [comment]
Merged as ingestion tests are passing.

**Pull request merged**